src/cargo/ops/cargo_compile/unit_generator.rs

```diff
@@ -267,35 +268,61 @@ impl<'a> UnitGenerator<'a, '_> {
                 })
                 .collect::<Vec<_>>();
             let suggestion = closest_msg(target_name, targets.iter(), |t| t.name(), "target");
-            if !suggestion.is_empty() {
-                anyhow::bail!(
-                    "no {} target {} `{}`{}",
-                    target_desc,
-                    if is_glob { "matches pattern" } else { "named" },
-                    target_name,
-                    suggestion
-                );
+            let target_elsewhere = if is_glob {
+                let pattern = build_glob(target_name)?;
+                let filter = |t: &Target| is_expected_kind(t) && pattern.matches(t.name());
+                self.get_targets_from_other_packages(filter)?
             } else {
-                let mut msg = String::new();
-                writeln!(
-                    msg,
-                    "no {} target {} `{}`.",
```

Comment:
unifying whether we have a `.` or not should probably be in its own commit.  I would lean instead to keep it instead of get rid of it

Reply:
I chose keeping the `.` , if it feels good to get rid of it then it will sperate into another refactor PR